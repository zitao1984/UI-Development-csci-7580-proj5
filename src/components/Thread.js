import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faTags} from "@fortawesome/free-solid-svg-icons";



const Thread=props=>{

    const time = new Date(props.thread.post_time)
    // console.log(props.thread)

    return (

        <div className="card text-dark bg-light mb-3" key={props.thread.id} style={{marginBottom:'1em' }}>
            <div className="card-header">Posted by: {time.toString()}</div>
            <div className="card-body">
                <h2 className="card-title">{props.thread.title}</h2>
                <h3 className="card-subtitle">From: {props.thread.id} </h3>
                <h4 className="card-text text-end"> <FontAwesomeIcon icon={faTags}/> {props.thread.tags.toString()}</h4>
                <p className="card-text mt-3 thread-p ">{props.thread.message}</p>
            </div>
        </div>
    )

}

export default Thread;