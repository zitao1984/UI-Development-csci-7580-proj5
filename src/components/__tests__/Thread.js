import Thread from "../Thread";
import renderer from "react-test-renderer";

const morckThread =
    {
        id: "mike",
        message: "haha",
        post_time: 1617686046238,
        tags: ["break"],
        thread_id: "1BHN9NL2tqRgHOxNbJja",
        title: "linebreak"
    }

test("The single thread render",()=>{
    const component = renderer.create(
        <Thread thread={morckThread}/>
    )

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})
