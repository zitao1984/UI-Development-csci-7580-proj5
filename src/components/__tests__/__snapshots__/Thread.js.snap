// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The single thread render 1`] = `
<div
  className="card text-dark bg-light mb-3"
  style={
    Object {
      "marginBottom": "1em",
    }
  }
>
  <div
    className="card-header"
  >
    Posted by: 
    Tue Apr 06 2021 00:14:06 GMT-0500 (Central Daylight Time)
  </div>
  <div
    className="card-body"
  >
    <h2
      className="card-title"
    >
      linebreak
    </h2>
    <h3
      className="card-subtitle"
    >
      From: 
      mike
       
    </h3>
    <h4
      className="card-text text-end"
    >
       
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-tags fa-w-20 "
        data-icon="tags"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
       
      break
    </h4>
    <p
      className="card-text mt-3 thread-p "
    >
      haha
    </p>
  </div>
</div>
`;
